pkg/gpu/nvidia/metrics/metrics.go

```diff
@@ -179,6 +205,23 @@ func (m *MetricServer) updateMetrics(containerDevices map[ContainerID][]string)
 			MemoryUsed.WithLabelValues(container.namespace, container.pod, container.container, "nvidia", d.UUID, *d.Model).Set(float64(*mem.Global.Used) * 1024 * 1024) // memory reported in bytes
 		}
 	}
+	node_name := os.Getenv("NODE_NAME")
+	for device, d := range gpuDevices {
+		status, err := gmc.collectStatus(d)
+		if err != nil {
+			glog.Errorf("Failed to get device status for %s: %v", device, err)
+		}
+		mem := status.Memory
+		dutyCycle, err := gmc.collectDutyCycle(d.UUID, time.Second*10)
+		if err != nil {
+			glog.Infof("Error calculating duty cycle for device: %s: %v. Skipping this device", device, err)
+			continue
+		}
```

Comment:
This part of the code is duplicated with existing code, do you think if we can put them into a private function?

Reply:
Done.